vm-setup/roles/v1aX_integration_test/tasks/move.yml

```diff
@@ -95,53 +107,49 @@
 
   # Check for pods & nodes on the target cluster
   - name: Check if pods in running state
-    shell: "kubectl get pods -A -o json | jq -r '.items[].status.phase' | grep -v Running"
-    environment:
-      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
-    retries: 150
+    k8s_info:
+      kind: pods 
+      kubeconfig: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+      field_selectors:
+        - status.phase!=Running
+    register: target_running_pods 
+    retries: 150 
     delay: 20
-    register: target_running_pods
-    failed_when: >
-      (target_running_pods.stderr != "") or
-      (target_running_pods.rc > 1) or
-      (target_running_pods.stdout != "")
-    until: target_running_pods.stdout == ""
-
+    until: target_running_pods.resources | length == 0
+    failed_when: target_running_pods.resources | length > 0 
+ 
   - name: Pivot objects to target cluster
     shell: "clusterctl move --to-kubeconfig /tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml -n {{ NAMESPACE }} -v 10"
 
   - name: Check if machines become running.
-    shell: |
-        kubectl get machines -n {{ NAMESPACE }} -o json | jq -r '[ .items[]
-        | select (.status.phase == "Running" or .status.phase == "running")
-        | .metadata.name ] | length'
-    register: provisioned_machines
-    environment:
-      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
-    retries: 50
+    k8s_info:
+      api_version: cluster.x-k8s.io/v1alpha3 
+      kind: machines 
+      kubeconfig: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    register: provisioned_machines 
+    retries: 50 
     delay: 20
-    until: provisioned_machines.stdout == NUMBER_OF_BMH
-
+    until: provisioned_machines | filter_running | length == NUMBER_OF_BMH | int
+  
   - name: Check if metal3machines become provisioned.
```

Comment:
the check is for being ready, can you change the name also accordingly

Reply:
Good catch